src/realm/table.hpp

```diff
@@ -2014,6 +2103,25 @@ inline void Table::set_ndx_in_parent(size_t ndx_in_parent) noexcept
     }
 }
 
+template <class T>
+void Table::set_list(size_t c, size_t r, const std::vector<T>& list)
```

Comment:
I don't think you example will compile :) 
I assume that `iList.get(2)` will somehow map to `List::operator[](2)`.
This List interface is just a thin wrapper around the (sub)table interface, so this will translate into `Table::get_int(0, 2)` which will succeed.